cartography/intel/azure/cosmosdb.py

```diff
@@ -0,0 +1,1090 @@
+    d.keyvaulturi = da.key_vault_key_uri,
+    d.enablefreetier = da.enable_free_tier,
+    d.enableanalyticalstorage = da.enable_analytical_storage,
+    d.defaultconsistencylevel = da.consistency_policy.default_consistency_level,
+    d.maxstalenessprefix = da.consistency_policy.max_staleness_prefix,
+    d.maxintervalinseconds = da.consistency_policy.max_interval_in_seconds
+    WITH d
+    MATCH (owner:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})
+    MERGE (owner)-[r:RESOURCE]->(d)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {azure_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest_database_account,
+        database_accounts_list=database_account_list,
+        AZURE_SUBSCRIPTION_ID=subscription_id,
+        azure_update_tag=azure_update_tag,
+    )
+
+
+@timeit
+def sync_database_account_data_resources(
+        neo4j_session: neo4j.Session, subscription_id: str, database_account_list: List[Dict], azure_update_tag: int,
+) -> None:
+    """
+    Loading all the resources (like cors policy, failover policy, private endpoint connections, virtual
+    network rules and locations) that we get as a part of the database account response itself.
+    """
+    for database_account in database_account_list:
+        if 'cors' in database_account and len(database_account['cors']) > 0:
+            db_account = transform_cosmosdb_cors_policy(database_account)
+            _load_cosmosdb_cors_policy(neo4j_session, db_account, azure_update_tag)
+        if 'failover_policies' in database_account and len(database_account['failover_policies']) > 0:
+            _load_cosmosdb_failover_policies(neo4j_session, database_account, azure_update_tag)
+        if 'private_endpoint_connections' in database_account and len(
+                database_account['private_endpoint_connections'],
+        ) > 0:
+            _load_cosmosdb_private_endpoint_connections(neo4j_session, database_account, azure_update_tag)
+        if 'virtual_network_rules' in database_account and len(database_account['virtual_network_rules']) > 0:
+            _load_cosmosdb_virtual_network_rules(neo4j_session, database_account, azure_update_tag)
+
+        locations = []
+        # Extracting every location
+        if 'write_locations' in database_account and len(database_account['write_locations']) > 0:
+            for loc in database_account['write_locations']:
+                locations.append(loc)
+        if 'read_locations' in database_account and len(database_account['read_locations']) > 0:
+            for loc in database_account['read_locations']:
+                locations.append(loc)
+        if 'locations' in database_account and len(database_account['locations']) > 0:
+            for loc in database_account['locations']:
+                locations.append(loc)
+        # Selecting only the unique location entries
+        loc = [i for n, i in enumerate(locations) if i not in locations[n + 1:]]
+        if len(loc) > 0:
```

Comment:
is this `if` necessary? Can you remove it and only have the loop?

Reply:
Oh yeah. Removed it.